tests/integration/helm/upgrade/helm_upgrade_test.go

```diff
@@ -16,171 +16,17 @@
-
-			ctx.ConditionalCleanup(func() {
-				// only need to do call this once as helm doesn't need to remove
-				// all versions
-				deleteIstio(cs, h)
-			})
-
-			overrideValuesFile := getValuesOverrides(ctx, defaultValues, gcrHub, previousSupportedVersion)
-			installIstio(t, cs, h, overrideValuesFile)
-			helmtest.VerifyInstallation(ctx, cs)
-
-			oldClient, oldServer := sanitycheck.SetupTrafficTest(t, ctx)
-			sanitycheck.RunTrafficTestClientServer(t, oldClient, oldServer)
-
-			// now upgrade istio to the latest version found in this branch
-			// use the command line or environmental vars from the user to set
-			// the hub/tag
-			s, err := image.SettingsFromCommandLine()
-			if err != nil {
-				ctx.Fatal(err)
-			}
-
-			overrideValuesFile = getValuesOverrides(ctx, defaultValues, s.Hub, s.Tag)
-			upgradeCharts(ctx, h, overrideValuesFile)
-			helmtest.VerifyInstallation(ctx, cs)
-
-			newClient, newServer := sanitycheck.SetupTrafficTest(t, ctx)
-			sanitycheck.RunTrafficTestClientServer(t, newClient, newServer)
-
-			// now check that we are compatible with N-1 proxy with N proxy
-			sanitycheck.RunTrafficTestClientServer(t, oldClient, newServer)
-		})
-}
-
-// upgradeCharts upgrades Istio using Helm charts with the provided
-// override values file to the latest charts in $ISTIO_SRC/manifests
-func upgradeCharts(ctx framework.TestContext, h *helm.Helm, overrideValuesFile string) {
-	// Upgrade base chart
-	err := h.UpgradeChart(helmtest.BaseReleaseName, filepath.Join(helmtest.ChartPath, helmtest.BaseChart),
-		helmtest.IstioNamespace, overrideValuesFile, helmtest.HelmTimeout)
-	if err != nil {
-		ctx.Fatalf("failed to upgrade istio %s chart", helmtest.BaseChart)
-	}
-
-	// Upgrade discovery chart
-	err = h.UpgradeChart(helmtest.IstiodReleaseName, filepath.Join(helmtest.ChartPath, helmtest.ControlChartsDir, helmtest.DiscoveryChart),
-		helmtest.IstioNamespace, overrideValuesFile, helmtest.HelmTimeout)
-	if err != nil {
-		ctx.Fatalf("failed to upgrade istio %s chart", helmtest.DiscoveryChart)
-	}
-
-	// Upgrade ingress gateway chart
-	err = h.UpgradeChart(helmtest.IngressReleaseName, filepath.Join(helmtest.ChartPath, helmtest.GatewayChartsDir, helmtest.IngressGatewayChart),
-		helmtest.IstioNamespace, overrideValuesFile, helmtest.HelmTimeout)
-	if err != nil {
-		ctx.Fatalf("failed to upgrade istio %s chart", helmtest.IngressGatewayChart)
-	}
-
-	// Upgrade egress gateway chart
-	err = h.UpgradeChart(helmtest.EgressReleaseName, filepath.Join(helmtest.ChartPath, helmtest.GatewayChartsDir, helmtest.EgressGatewayChart),
-		helmtest.IstioNamespace, overrideValuesFile, helmtest.HelmTimeout)
-	if err != nil {
-		ctx.Fatalf("failed to upgrade istio %s chart", helmtest.EgressGatewayChart)
-	}
-}
-
-// installIstio install Istio using Helm charts with the provided
-// override values file and fails the tests on any failures.
-func installIstio(t *testing.T, cs cluster.Cluster,
```

Comment:
Moved all unexported functions (which were similar in both packages) to the "upgrade" package and made them exported. Code was then moved to `util.go`